src/QsCompiler/QirGeneration/Subtransformations/ExpressionKindTransformation.cs

```diff
@@ -386,6 +386,252 @@ IValue CopyAndUpdateUdt(TupleValue originalValue)
             return ResolvedExpressionKind.InvalidExpr;
         }
 
+        /// <summary>
+        /// Creates a string literal that contains the given values as interpolation arguments.
+        /// The given string is expected to consist of text interspaced with integer constants between curly brackets.
+        /// The content between non-escaped curly brackets is parsed as integer,
+        /// and replaced with the string representation of the interpolation argument at that index.
+        /// Registers the built string with the scope manager.
+        /// </summary>
+        internal static IValue CreateStringLiteral(GenerationContext sharedState, string str, params IValue[] exs)
+        {
+            static (int, int, int) FindNextExpression(string s, int start)
+            {
+                while (true)
+                {
+                    var i = s.IndexOf('{', start);
+                    if (i < 0)
+                    {
+                        return (-1, s.Length, -1);
+                    }
+                    else if ((i == start) || (s[i - 1] != '\\'))
+                    {
+                        var j = s.IndexOf('}', i + 1);
+                        if (j < 0)
+                        {
+                            throw new FormatException("Missing } in interpolated string");
+                        }
+                        var n = int.Parse(s[(i + 1)..j]);
+                        return (i, j + 1, n);
```

Comment:
Minor:
Just in case I want to make sure that your code, upon the call `int.Parse(..)`, correctly handles the cases when
`s` contains "{}" (char `}` immediately follows `{`), i.e. `i + 1` is equal to `j` in the fragment `s[(i + 1)..j]`;
`s` contains "{\<non Int>}", i.e. there is some identifier, or double value, etc. between the `{` and `}`.

I expect the call [`Parse(String)`](https://docs.microsoft.com/en-us/dotnet/api/system.int32.parse?view=net-5.0#System_Int32_Parse_System_String_) to throw certain exceptions in that case. So I wanted to make sure that those exceptions are handled properly.

Reply:
Throwing an exception in this case is expected, and acceptable, see my comment above regarding relying on the correctness of the AST.